src/nvim/eval/typval_encode_types.h

```diff
@@ -0,0 +1,59 @@
+/// @file eval/typval_encode_types.h
```

Comment:
We use `_defs.h` suffix (not `_types.h`) in other cases, so let's stay consistent with that.